agents/sbd/fence_sbd.py

```diff
@@ -356,6 +357,34 @@ def define_new_opts():
         "order": 200
         }
 
+
+def sbd_daemon_is_running():
+    """Check if the sbd daemon is running
+    """
+    if not os.path.exists(SBD_PID_FILE):
+        logging.info("SBD PID file %s does not exist", SBD_PID_FILE)
+        return False
+
+    try:
+        with open(SBD_PID_FILE, "r") as pid_file:
+            pid = int(pid_file.read().strip())
```

Comment:
Well we're opening the file in read-only and the garbage collector will finally destroy the file-object which in turn should close the file.
Not sure what is good practice in this case but as we don't know when the garbage collector will do this we might close explicitly.

Reply:
The `with` statement ensures the file is closed immediately after the block is executed, so we don't need to worry about when the garbage collector will run

That's the best practice by using the `with` statement